### PR TITLE
remove internal reallocarray()

### PR DESCRIPTION
**Pull request by vcunat**
~For reference, we ran into this problem due to strict sandboxing in [a packaging solution](https://github.com/nixos/nixpkgs).~

... as it's covered by libbpf_reallocarray() since commit dc70da9c70.

### CODE REVIEW

**Comment by anakryiko** [comment]
libbpf doesn't require check-reallocarray.sh anymore, would you mind instead just removing it?

**Comment by vcunat** [comment]
I certainly wouldn't mind.   I ran into 0.1.0 failing due to `/tmp` being blocked by sandbox and I actually haven't even checked against master (given the freshness of the release).

**Comment by anakryiko** [comment]
@vcunat, do you plan to do a follow up and remove check-reallocarray.sh altogether?

**Comment by vcunat** [comment]
Ah, I thought you/someone planned to do it, but now I see that it wasn't clear from your post.  So, I did not plan it, but I think I could do it within a couple days.

**Comment by vcunat** [comment]
I expect you wanted something like this?

File: src/libbpf_internal.h [hunk 1]
```diff
@@ -83,11 +83,8 @@ do {				\
 #endif
 /*
  * Re-implement glibc's reallocarray() for libbpf internal-only use.
- * reallocarray(), unfortunately, is not available in all versions of glibc,
- * so requires extra feature detection and using reallocarray() stub from
- * <tools/libc_compat.h> and COMPAT_NEED_REALLOCARRAY. All this complicates
- * build of libbpf unnecessarily and is just a maintenance burden. Instead,
- * it's trivial to implement libbpf-specific internal version and use it
```
Comment (anakryiko) [inline src/libbpf_internal.h:90]: please don't change this file, it gets auto-synced

File: scripts/sync-kernel.sh [hunk 2]
```diff
@@ -45,7 +45,6 @@ PATH_MAP=(									\
 	[tools/include/uapi/linux/if_link.h]=include/uapi/linux/if_link.h	\
 	[tools/include/uapi/linux/if_xdp.h]=include/uapi/linux/if_xdp.h		\
 	[tools/include/uapi/linux/netlink.h]=include/uapi/linux/netlink.h	\
-	[tools/include/tools/libc_compat.h]=include/tools/libc_compat.h		\
```
Comment (anakryiko) [inline scripts/sync-kernel.sh:48]: can you please remove this file from include/tools as well?

**Review by anakryiko — COMMENTED** [review]
almost :)

File: scripts/sync-kernel.sh (hunk 2 above)
Comment (vcunat) [reply scripts/sync-kernel.sh:48]: I did remove `include/tools/libc_compat.h`... so you missed that or I don't understand?

File: src/libbpf_internal.h (hunk 1 above)
Comment (vcunat) [reply src/libbpf_internal.h]: Ah, I wasn't watching out for that.  Amended.

File: scripts/sync-kernel.sh (hunk 2 above)
Comment (anakryiko) [reply scripts/sync-kernel.sh:48]: ah, my bad, missed it in "Files changed" view, sorry about that

Comment (vcunat) [reply scripts/sync-kernel.sh:48]: No problem :-)

**Pull request merged**